Common/Gfx/Canvas.cpp

```diff
@@ -563,6 +563,34 @@ void Canvas::DrawMaskedBitmap(Gdiplus::Bitmap* bitmap, Gdiplus::Bitmap* maskBitm
 	bitmapLock->Release();
 }
 
+Microsoft::WRL::ComPtr<ID2D1RectangleGeometry> Canvas::CreateRectangle(D2D1_RECT_F rectangle)
+{
+	Microsoft::WRL::ComPtr<ID2D1RectangleGeometry> c_Geometry = NULL;
+	HRESULT hr = c_D2DFactory->CreateRectangleGeometry(rectangle, &c_Geometry);
+	return c_Geometry;
+}
+
+void Canvas::DrawGeometry(const Shape & shape, D2D1_MATRIX_3X2_F & transform)
+{
+	if (!BeginTargetDraw()) return;
+
+	D2D1_MATRIX_3X2_F worldTransform;
+	m_Target->GetTransform(&worldTransform);
+	m_Target->SetTransform(transform * worldTransform);
+	Microsoft::WRL::ComPtr<ID2D1SolidColorBrush> solidBrush = nullptr;
```

Comment:
I've always thought that it was a good thing to initalize it to null so that it doesn't take some garbage value. It isn't needed tho so i'll remove it.